Sharpmake.Generators/VisualStudio/Vcxproj.cs

```diff
@@ -854,7 +854,7 @@ Strings ignoreSpecificLibraryNames
 
             // put the built library files before any other
             var libraryFiles = new OrderableStrings(context.Configuration.DependenciesBuiltTargetsLibraryFiles);
-            libraryFiles.Sort();
+            // libraryFiles.Sort(); // Note: Leave libraries in depedency order for llvm platforms
```

Comment:
We need to sort libraries. If for some reason it doesn't work to use the TargetFileOrderNumber it needs fixing

Reply:
Oh, I didn't notice that the Strings has the order number built in. 

I also ran into this a while ago when trying to use Make files as well, however I was able to inject gcc -start-group -end-group to get around that. I don't have that option here unless I move all the libs to additional linker options.

Do you think it would be ok to try to auto gen the TargetFileOrderNumber for dependency includes if it's not specified by the user?

Reply:
Updated the sorting to be deterministic and opt-in.